### PR TITLE
fix(carousel): Handle changes in slide content

### PR DESCRIPTION
**Pull request by tmorehouse**
Add `observedom` mixin to handle updates to slide content (and delayed rendering of `carousel-slide`)

No guarantee that carousel is bug-free

### CODE REVIEW

**Comment by pi0** [comment]
I have temporarily removed carousel from docs sidebar. Do you think we are stable enough now to bring it back?

**Comment by tmorehouse** [comment]
Now includes fixes for slightly messed up transitions (blank space between sliding slides is now gone)

Allows user to programmatically pause and restart the carousel by setting `interval` prop on the fly

**Comment by tmorehouse** [comment]
I think this version fixes many issues (although not positive if it has added any new "bugs")

**Comment by tmorehouse** [comment]
Also added updated docs/example for carousel

File: lib/components/carousel.vue [hunk 1]
```diff
@@ -139,27 +140,35 @@
             },
             value: {
                 type: Number,
-                defrault: 0
```
Comment (tmorehouse) [inline lib/components/carousel.vue]: This was the main culprit of carousel not showing initially, as value defaulted to `null`/`undefined` if a v-model wasn't bound.

**Comment by tmorehouse** [comment]
This should now be ready for testing before merge.

**Comment by tmorehouse** [comment]
Incorporated changes by @pi0 in the v4-beta branch

**Pull request merged**